File: mdd-back/src/main/java/com/openclassroom/mdd_app/entities/Article.java
package com.openclassroom.mdd_app.entities;

import java.time.LocalDateTime;
import java.util.List;

import org.hibernate.annotations.CreationTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;


@Data
@Entity
@Table(name="articles")
public class Article {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;
	
	@NotNull(message = "title cannot be NULL")
	@Column(nullable = false)
	private String title;
	
	@Size(max=2500)
	private String description;
	
	@CreationTimestamp
	private LocalDateTime date;
	
	@NotNull
	@ManyToOne
	@JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
	private User user;

	@NotNull
	@ManyToOne
	@JoinColumn(name = "topic_id", referencedColumnName = "id", nullable = false)
	private Topic topic;
	
	@OneToMany(mappedBy = "article")
    private List<Comment> comments;
}

File: mdd-front/src/app/models/article.ts
import { Topic } from './topic';
import { Comment } from './comment';

export class Article {
  id!: number;
  title!: string;
  description!: string;
  date!: string;
  user!: UserInfo;
  topic!: Topic;
  comments!: [];
}

export class UserInfo {
  id!: number;
  username!: string;
}

export class ArticleRequest {
  title!: string;
  description!: string;
  topicId!: number;
}

export class ArticleResponse {
  id!: number;
  title!: string;
  description!: string;
  userId!: number;
  topicId!: number;
}

export class ArticleDetail {
  id!: number;
  title!: string;
  description!: string;
  date!: string;
  user!: UserInfo;
  topic!: Topic;
  comments!: Comment[];
}
